### ISSUE TITLE
[dev.icinga.com #7968] Provide a mechanism to request an update of the content instead of updating automatically

### ISSUE THREAD

**Issue by icinga-migration** [Icinga/icingaweb2#1239]
This issue has been migrated from Redmine: https://dev.icinga.com/issues/7968

**Created by mhein on 2014-12-03 11:36:23 +00:00**

Assignee: _afuhr_
Status: _Resolved_  (closed on _2015-02-25 13:29:24 +00:00_)
Target Version: _2.0.0-beta3_
Last Update: _2015-02-25 13:29:24 +00:00 (in Redmine)_

---

http://www.w3.org/TR/WCAG20-TECHS/G76.html wrote:
> The objective of this technique is to let the user control if and when content is updated, in order to avoid confusion or disorientation caused by automatic refreshes that cause a change of context. Users of screen readers may find automatic updates confusing because it is not always clear what is happening. When a page is refreshed, the screen reader's “virtual cursor", which marks the user's current location on the page, is moved to the top of the page. People who use screen magnification software and people with reading disabilities may also be disoriented when pages are refreshed automatically. \[...\]

Next to our dropdown tab should be a refresh control.

**Changesets**

_2015-02-23 16:01:38 +00:00_ by _afuhr_ ac26807ab788e6fdc100dddcedbce231ac3b2347

    Add new spin an cw icon for refresh control
    
    refs #7968

_2015-02-23 16:02:44 +00:00_ by _afuhr_ aa5c52dfdded32dc9f4248d06395794cf9e716b7

    Implement content refresh control next to dropdown tab
    
    refs #7968

_2015-02-25 11:07:59 +00:00_ by _afuhr_ 3b2e3dcd36728c8ddc1b3b4f872ca70fa9bcc4e6

    Revert the animate-spin css to the original
    
    refs #7968

_2015-02-25 11:10:08 +00:00_ by _afuhr_ 4505718ecc22241d967cbe85298d2ee7aa161421

    Move spinner indicator functionality to linkClicked method
    
    refs #7968

_2015-02-25 11:11:58 +00:00_ by _afuhr_ c583a351bb5406ef9e3204e3d2eda74b9ca4c65a

    Add .spinner css definition
    
    refs #7968

_2015-02-25 11:14:26 +00:00_ by _afuhr_ 481c58ec039917ae7d1368a60f93bbd3bdb223c6

    Add accessibility to renderRefreshTab
    
    refs #7968

_2015-02-25 12:13:22 +00:00_ by _afuhr_ e219bb1664f925668c8a7eb71b78cb4a5577f6c7

    Add doc for the label
    
    refs #7968

_2015-02-25 12:23:12 +00:00_ by _afuhr_ 8669dcb078952ed6f3192ba861f5c1e0a9ffbac6

    Add font-family to the active spinner icon
    
    refs #7968

_2015-02-25 12:24:11 +00:00_ by _afuhr_ 3a2b69a63ceedeb5e48c354e36eb74e5db11026b

    Remove refreshTab property from Tabs
    
    refs #7968

_2015-02-25 12:34:58 +00:00_ by _afuhr_ ef79d6bc12ba85f27cd39cf37acc2eb2cf0bce38

    Replace spinner.active font-family css definition with a note
    
    refs #7968

_2015-02-25 12:51:26 +00:00_ by _afuhr_ 7f883a47701dd4834258359ddeafff26b71846e9

    Add own animation.less to provide own parameterized animate function
    
    refs #7968

_2015-02-25 12:53:00 +00:00_ by _afuhr_ c029ec1af92abc86d03bac0f93ed00aef8e06fb4

    Replace animate-spin with own animate implementation in spinner
    
    refs #7968

_2015-02-25 12:56:32 +00:00_ by _afuhr_ 08ace05ac750ce54810648ea202dd8c5f7d97904

    Remove animations.css from StyleSheet loading lessFiles
    
    refs #7968

_2015-02-25 13:15:32 +00:00_ by _afuhr_ 16200417ebb449869057c41cfd8520ea2e64c1e7

    Add spinner active indicator for form submit buttons
    
    refs #7968

_2015-02-25 13:16:35 +00:00_ by _afuhr_ 797d37735d3ddfe775bb9d390234e529685ac873

    Add .spinner class to CheckNowCommandForm submit button
    
    refs #7968

_2015-02-25 13:28:25 +00:00_ by _afuhr_ 453e0af9878c6f877fa2978d46aec28cf06630fe

    Merge branch 'feature/refresh-control-7968'
    
    resolves #7968

_2015-02-25 14:28:09 +00:00_ by _afuhr_ cc403806f764c1527afc78cc451b4745a945d475

    Fix refresh accessibility lable if it is empty
    
    refs #7968
